File: binarySearch.js
function swap(array, firstIndex, secondIndex){
	var temp = array[firstIndex];
	array[firstIndex] = array[secondIndex];
	array[secondIndex] = temp;
}
function binarySearch(array, target){
	var min = 0;
	var max = array.length-1;
	var guess;
	while(min <= max){
		guess = Math.floor((min + max) / 2);
		if(array[guess] > target){
			max = guess - 1;
		} else if(array[guess] < target){
			min = guess + 1;
		} else {
			return guess;
		}
	}
	return -1;
}

var arr = [0, 2, 4, 12, 21, 88, 101, 128, 144, 255];
console.log(binarySearch(arr, 252));

File: BinarySearch.java
class BinarySearch {
	
	private static int min;
	private static int mid;
	private static int max;
	
	public static int search(int[] arr, int target){
		min = 0;
		max = arr.length-1;
		while(min <= max){
			mid = (min + max) / 2;
			if(arr[mid] > target)	max = mid - 1;
			else if(arr[mid] < target) min = mid + 1;
			else	return mid;
		}
		return -1;
	}
	
	public static void main(String[] args) {
		int[] arr = {-41, -16, 0, 3, 21, 33, 45, 77, 100, 300, 812};
		System.out.println(search(arr, 813));
	}
}